src/interface/viewport.c

```diff
@@ -270,15 +270,56 @@ void sub_689174(sint16* x, sint16* y, sint16 *z)
 	*z = height;
 }
 
-// ax = viewport->x
-// bx = viewport->y;
-// cx = viewport->width;
-// dx = viewport->height;
+void sub_683326(int left, int top, int right, int bottom)
+{
+	RCT2_CALLPROC_X(0x00683359, left, top, right, bottom, 0, 0, 0);
+}
+
+// shifts pixels from the region in a direction. Used when a viewport moves;
+// consider putting in src/drawing/drawing.c or src/drawing/rect.c
+// 
+// 0x00683359
+// ax = x
+// bx = y;
+// cx = width;
+// dx = height;
 // di = dx;
 // si = dy;
-void sub_683359(int x, int y, int width, int height, int dx, int dy)
+void gfx_move_screen_rect(int x, int y, int width, int height, int dx, int dy)
 {
-	RCT2_CALLPROC_X(0x00683359, x, y, width, height, dy, dx, 0);
+	// nothing to do
+	if (dx == 0 && dy == 0)
+		return;
+
+	// I have seen no advantages of having this function nor disadvantages from
+	// leaving it out. Kept in for completeness
+	sub_683326(x, y, x + width - 1, y + height - 1);
```

Comment:
@IntelOrca removed another instance of this function https://github.com/OpenRCT2/OpenRCT2/commit/4f77cc8df3cd9af35848d21a1234039ca0c02732
It looks to be something OS level since the function it calls is at 0x40YYYY which are generally C functions that are used to call os functions.